CMakeLists.txt

```diff
@@ -309,6 +309,14 @@ include_directories(SYSTEM ${LIBUV_INCLUDE_DIRS})
 find_package(Msgpack 1.0.0 REQUIRED)
 include_directories(SYSTEM ${MSGPACK_INCLUDE_DIRS})
 
+option(PREFER_LUAJIT "Prefer LuaJIT over Lua when compiling executable. Test library always uses luajit." ON)
+
+find_package(LuaJit REQUIRED)
```

Comment:
If I don't care to run tests, then this `REQUIRED` prevents me from being able to use `cmake -DPREFER_LUAJIT=OFF` to simply build nvim with Lua, doesn't it?  Shouldn't we instead avoid creating the relevant test targets if luajit isn't available?